### ISSUE TITLE
Mail.app resizing

### ISSUE THREAD

**Issue by Aqueum** [eczarny/spectacle#444]
When trying to position Mail.app Inbox windows to 1/2 width (1/2 screen or 1/4 screen) on a 15" screen Mail.App resizes itself after Spectacle's resize such that the resulting window is smaller than 1/2 width (empty space on each side).  

This seems to be because it has two minimum window widths - one uses a standard size search box, and one uses a smaller search box (or none if you have icon text showing).  If 1/2 screen with is less than the size of the larger minimum, Mail.App's minimum width is reduced AFTER Spectacle resizes to 1/2 screen.

Removing icons, or the search box from Mail.app's toolbar annoyingly does not appear to change the two minimum window sizes.

This means Spectacle 1.0 is still useless with Mail.app as I need to manually resize the inbox every time.  Spectacle 0.6.x managed to work fine with Mail.app, I first noticed this as an issue with Spectacle 0.8.10

I am experiencing this on a 15" MBP with retina display (late 2013 2.3GHz i7, 16GB RAM) running Spectacle 1.0 on OS X Yosemite 10.10.5.

**Comment by eczarny** [comment]
Is Spectacle behaving the same as it has before? Unfortunately there is little that can be done for windows with restricted dimensions.

**Comment by Aqueum** [comment]
Hi Eric

Thanks for checking.

Omnifocus and Skype seem to be fixed, one new issue is that every window seems to stop a few pixels above the bottom of the screen.

Mail.app is broken just as before.  Two questions on that:
Why did it work fine in 0.6?
Might it be possible to introduce an optional delay between resizing and repositioning windows.  I suspect that if Mail.App inbox was resized, then time allowed for it to resize itself, then it’s extremity was aligned with the relevant screen edge then it should all work fine.

Note that restricted dimensions is not a problem in my mind, I don’t mind if a ½ sized app goes beyond the half way, as long as it lines up with the chosen screen edge - in fact Mail.app is smaller than ½ sized, the annoyance is the waste of space in the gap between the screen edge and its extremity.

Kind regards

Martin

---
Martin Currie
Aqueum
UK: +44 20 8144 1024
US: +1 646 419 4102
Mauritius: +230 5987 1023

**Comment by eczarny** [comment]
If your Dock is hidden on the bottom of the screen this could explain the new issue you are seeing. I would love to see a screenshot though.

I did make some changes in 1.0 that should align windows with restricted dimensions to the edges of the screen. It's hard to say why this worked in 0.6 but my guess is Apple has made changes to Mail.app since that could be problematic.

**Comment by Aqueum** [comment]
Thank Eric

Please see the screen shot attached.  You can see the small gap, which I agree is due to the hidden dock - it seems that it doesn’t happen if you first activate the dock on the other screen…

On the right window you’ll see the gaps that I mean with the Mail.app inbox.  I think you’re not quite understanding the problem there - it is not that the windows have restricted dimensions, its that they have TWO sets of restricted dimensions one above a certain width and one below.  Spectacle aligns for the larger restricted dimension, but the window then resizes itself to the smaller.



Kind regards

Martin

---
Martin Currie
Aqueum
UK: +44 20 8144 1024
US: +1 646 419 4102
Mauritius: +230 5987 1023

**Comment by eczarny** [comment]
Sorry, I don't see any screenshot.

**Comment by Aqueum** [comment]
Eric

I have attached the screenshot again (a .png), but wouldn’t be surprised if github strips images off - is there a regular email address I can send it to if it doesn’t reach you this time?



Kind regards

Martin
---
Martin Currie
Aqueum
UK: +44 20 8144 1024
US: +1 646 419 4102
Mauritius: +230 5987 1023

**Comment by eczarny** [comment]
Try attaching the image to a comment on github.com instead of using email.

**Comment by Aqueum** [comment]
I’d rather not publicly post a screen shot, can you access it here: https://dl.dropboxusercontent.com/u/5419314/Screen%20Shot%202015-09-02%20at%2010.50.44.png <https://dl.dropboxusercontent.com/u/5419314/Screen%20Shot%202015-09-02%20at%2010.50.44.png>

---
Martin Currie
Aqueum
UK: +44 20 8144 1024
US: +1 646 419 4102
Mauritius: +230 5987 1023

**Comment by eczarny** [comment]
If your Dock is hidden on the bottom that would explain the gap.

**Comment by Aqueum** [comment]
Agreed, that’s clearly the problem at the bottom - but the issue is the gaps around the sides of the leftmost window.

Kind regards

Martin

---
Martin Currie
Aqueum
UK: +44 20 8144 1024
US: +1 646 419 4102
Mauritius: +230 5987 1023

**Comment by eczarny** [comment]
Oh, I see. Interesting. I think you're seeing some special behavior that Spectacle uses as a fallback for apps like Terminal and iTerm 2. Mail.app does some weird things when you make a window smaller; not sure what to do in this particular case.

**Comment by Aqueum** [comment]
Might it be possible to introduce an optional delay between resizing and repositioning windows.  I suspect that if Mail.App inbox was resized, then time allowed for it to resize itself, then it’s extremity was aligned with the relevant screen edge then it should all work fine.
---
Martin Currie
Aqueum
UK: +44 20 8144 1024
US: +1 646 419 4102
Mauritius: +230 5987 1023

**Comment by eczarny** [comment]
Delays rarely help in cases like this. They tend to just make things more complicated and less reliable.

**Comment by eczarny** [comment]
@Aqueum I'm curious if this issue still exists on El Capitan with Spectacle 1.0.6.

**Comment by Aqueum** [comment]
Thanks for checking up Eric.

The original issue seems to be resolved, there's a funky little jiggle that makes it look like the windows are physically banging into place.  I actually quite like that.  But once they land we're not getting the annoying gaps that we had previously.

The bottom gap with a hidden doc still seems to be present.
